wingtips-core/src/main/java/com/nike/wingtips/http/HttpRequestTracingUtils.java

```diff
@@ -41,6 +41,32 @@
      */
     public static final String UNSPECIFIED_SPAN_NAME = "UNSPECIFIED";
 
+    /**
+     * A special indicator tag key (with an expected value of "true") that will be added to a span by {@link
+     * #fromRequestWithHeaders(RequestWithHeaders, List)} when the caller sent a trace ID but not a span ID, and
+     * therefore the {@link Span#getSpanId()} on the span returned by {@link
+     * #fromRequestWithHeaders(RequestWithHeaders, List)} is random and cannot be trusted.
+     *
+     * <p>See {@link #hasInvalidSpanIdBecauseCallerDidNotSendOne(Span)} for a helper that can be used to determine
+     * if a span falls into this category.
+     */
+    public static final String SPAN_FROM_HEADERS_WHERE_CALLER_DID_NOT_SEND_SPAN_ID_TAG_KEY =
+        "span_from_headers_caller_did_not_send_span_id";
+
+    /**
+     * The direct children of a span from {@link #fromRequestWithHeaders(RequestWithHeaders, List)} where the caller
+     * did not send a span ID (see {@link #hasInvalidSpanIdBecauseCallerDidNotSendOne(Span)}) should apply a tag with
+     * this key and a value of "true" to indicate that the {@link Span#getParentSpanId()} is invalid and cannot
+     * be trusted. It's recommended that the parent span ID should be set to null in this case - this can lead to
+     * multiple roots in your trace tree, but is the most reasonable way to handle the situation without completely
+     * breaking the trace and starting a new one.
+     *
+     * <p>See {@link #hasInvalidParentIdBecauseCallerDidNotSendSpanId(Span)} for a helper that can be used to determine
+     * if a span falls into this category.
+     */
+    public static final String CHILD_OF_SPAN_FROM_HEADERS_WHERE_CALLER_DID_NOT_SEND_SPAN_ID_TAG_KEY =
```

Comment:
this sounds horrible. also it can mask propagation that can only send a standalone traceId (ex aws can do this). Is this an internal bug? If not, it should be loudly escalated and fixed rather than worked around.

Reply:
@adriancole We have clients that only send a trace ID. This is being fixed where possible, but it's not a quick process.

This PR attempts to improve behavior when clients do this and make the issue easier to discover and act on, not make things worse. I'm hoping there's just a misunderstanding here.

Current behavior:
* Client sends a request with only trace ID.
* Server creates a new span with:
    + The same trace ID. 
    + A random parent span ID that doesn't point to anything real.
* Result - the server span is an orphan, pointing to a parent that doesn't exist and will never exist.

Behavior after this PR:
* Client sends a request with only trace ID.
* Server creates a new span with:
    + The same trace ID.
    + A null parent span ID.
    + A tag indicating that the caller only sent trace ID.
* Result - the server span has a null parent span ID, which could lead to multiple roots in the trace tree. But it also will include a tag explaining why the parent span ID is null, so you can search/filter on this scenario. This seems like a much better place to be than pointing to a random parent span ID that doesn't really exist, and with no way to know that the caller is at fault.

(Note that the span created from the `fromRequestWithHeaders(...)` method is never used, except as a synthetic parent when creating the server span. It's never completed or recorded. It's discarded as soon as the server span is generated. I would agree it's clunky.)

> also it can mask propagation that can only send a standalone traceId (ex aws can do this)

I'm not sure what you mean here. Surely the previous behavior of the server span pointing to a random parent that doesn't exist is worse than a server span with a null parent, and with a tag explaining the situation? 

I'm hoping you misunderstood what this PR was doing, and I'm not making things worse. Let me know if you get a chance. Thanks!